Add API and tests for .children()

abuuzayr opened the pull request:
Adds methods for `.children()` and appropriate tests.

Closes #1

abuuzayr commented:
Ready for your review @franciscop ..

Thank you for your time !

franciscop commented:
Thanks! I can review it better in few hours, but from a quick glance it looks great! Couple of improvements:

- It should probably return an instance of $ to be able to chain it: `$(...).children().find('...')`
- I see the code is doing `.first()` only, is this on purpose? Would it be better to do `this.nodes.map(...)` instead of only getting the children of the first one?

abuuzayr commented:
Thanks for the quick review! 

That makes sense. I updated:

1) Test for chaining
2) Converted to get multiple children instead of just the first one

franciscop commented on src/index.js:
```diff
@@ -58,4 +58,16 @@ $.prototype.click = function(...args) {
+      }
+    })
+  })
+  return $(children);
```
Great chaining! 👍

franciscop commented on src/index.js:
```diff
@@ -58,4 +58,16 @@ $.prototype.click = function(...args) {
+    node.childNodes.forEach(childNode => {
+      if (!selector || childNode.matches(selector)) {
+        children.push(childNode);
+      }
```
Looks good, I'd normally split that into 2 statements to make it clear those are 2 totally different scenarios and not part of a single one, but since they are simple enough it should be fine 👍

abuuzayr replied:
Thanks! I think that makes the scenarios more obvious. I'll put this in the refactor since I'm making changes anyway.

franciscop commented on index.min.js:
```diff
@@ -79,6 +79,19 @@
+      }
+    });
+    return $(children);
+  };
```
Needs to be re-built for this, though I do that before publishing a new version so not an issue

franciscop commented:
Also restructured things around so that each part is self-contained in a single folder and added a dependency, so if you pull to add `.children()` in its own folder you'll have to `npm install` again before `npm run build`.

abuuzayr commented:
Wow thanks @franciscop.. I've never been a collaborator before, I hope I don't break anything! 

Noted on the re-structure, I'll make the amendments and ping you when its done. If I am unable to resolve the changes automatically, I'll just re-open a new PR and tag you there. 

Thank you!

franciscop commented on src/children/index.js:
```diff
@@ -0,0 +1,16 @@
+      if (!selector) {
+        children.push(childNode);
+      }
+      if (childNode.matches(selector)) {
```
Do we need `else if` here?

abuuzayr replied:
I think that would make it more intuitive, yes.

abuuzayr commented:
Awesome, merging!
